src/OrchardCore.Modules/OrchardCore.Users/Drivers/RegisterUserFormDisplayDriver.cs

```diff
@@ -42,11 +42,11 @@ public override async Task<IDisplayResult> UpdateAsync(RegisterUserForm model, U
 
         await context.Updater.TryUpdateModelAsync(vm, Prefix);
 
-        if (await _userManager.FindByNameAsync(vm.UserName) != null)
+        if (!string.IsNullOrEmpty(vm.UserName) && await _userManager.FindByNameAsync(vm.UserName) != null)
```

Comment:
Why not decorate the VM property with `Required` attribute instead?

Reply:
the VM is actually is decorated. but it could be empty there. FYI, I do not like to decorate the model with the attribute. I rather see the validation done in the driver since that would show you validation in one place.

Reply:
I know. but it does not make sense to me to have validation in two places as we do here. I am okay with data annotation alone but do not like to have it into separate places. Any how, that discussion goes beyond this PR

Reply:
No problem let's move the check before line 45 so no need to check twice

Reply:
You mean remove data annotation? I rather not do that because currently there are tests against them. After merging this PR, feel free to submit a refactoring PR that would remove the data annotation and fix any tests that may break

Reply:
Ok I will do after you PR, let us now check the username before to avoid checking it twice

Reply:
I don't understand. In the driver we check the username (once) and the email (once).

Reply:
You're right I thought we are checking the same thing